functional-samples/tutorial.mole-game/mole/service-worker.js

```diff
@@ -0,0 +1,38 @@
+let failTimeout;
+let moleShowing = false;
+let controllerId;
+
+chrome.action.onClicked.addListener(() => {
+  if (!moleShowing) return;
+
+  chrome.runtime.sendMessage(controllerId, 'success');
+  hideMole();
+
+  failTimeout && clearTimeout(failTimeout);
+  failTimeout = undefined;
+});
+
+function showMole() {
+  chrome.action.setIcon({ path: 'icon-mole.png' });
+  moleShowing = true;
+}
+
+function hideMole() {
+  chrome.action.setIcon({ path: 'icon-empty.png' });
+  moleShowing = false;
+}
+
+chrome.runtime.onMessageExternal.addListener((msg) => {
+  controllerId = msg.id;
+  showMole();
+  failTimeout = setTimeout(async () => {
+    hideMole();
+    const tabs = await chrome.tabs.query({});
+    const eligibleTabs = tabs.filter((t) => t.title.includes('Example'));
```

Comment:
Why should the title include "example"?

Reply:
Ah, good spot - I'll get that fixed. That was there when testing to avoid losing my own tabs.